contracts/services/ibc-transfer/src/contract.rs

```diff
@@ -0,0 +1,213 @@
+) -> Result<Response, ServiceError> {
+    valence_service_base::instantiate(deps, CONTRACT_NAME, CONTRACT_VERSION, msg)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg<ActionsMsgs, OptionalServiceConfig>,
+) -> Result<Response<NeutronMsg>, ServiceError> {
+    valence_service_base::execute(
+        deps,
+        env,
+        info,
+        msg,
+        actions::process_action,
+        execute::update_config,
+    )
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(deps: DepsMut, env: Env, msg: Reply) -> StdResult<Response> {
+    if valence_ibc_utils::neutron::is_ibc_transfer_reply(&msg) {
+        return valence_ibc_utils::neutron::handle_ibc_transfer_reply::<Empty>(deps, env, msg);
+    }
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn sudo(deps: DepsMut, env: Env, msg: SudoMsg) -> StdResult<Response> {
+    if valence_ibc_utils::neutron::is_ibc_transfer_sudo(&msg) {
+        return valence_ibc_utils::neutron::handle_ibc_transfer_sudo(
+            deps,
+            env,
+            msg,
+            sudo::ibc_transfer_sudo_callback,
+        );
+    }
+    Ok(Response::default())
+}
+
+mod actions {
+    use cosmwasm_std::{DepsMut, Empty, Env, MessageInfo, Response, SubMsg, Uint128};
+    use neutron_sdk::bindings::{msg::NeutronMsg, query::NeutronQuery};
+    use valence_service_utils::{error::ServiceError, execute_on_behalf_of};
+
+    use crate::msg::{ActionsMsgs, Config};
+
+    pub fn process_action(
+        deps: DepsMut<NeutronQuery>,
+        env: Env,
+        _info: MessageInfo,
+        msg: ActionsMsgs,
+        cfg: Config,
+    ) -> Result<Response<NeutronMsg>, ServiceError> {
+        match msg {
+            ActionsMsgs::IbcTransfer {} => {
+                let balance = cfg.denom().query_balance(&deps.querier, cfg.input_addr())?;
+                if balance < *cfg.amount() {
+                    return Err(ServiceError::ExecutionError(format!(
+                        "Insufficient balance for denom '{}' in config (required: {}, available: {}).",
+                        cfg.denom(), cfg.amount(), balance,
+                    )));
+                }
+
+                // Send funds from input account to the IBC transfer service
+                let transfer_msg = cfg
+                    .denom()
+                    .get_transfer_to_message(&env.contract.address, *cfg.amount())?;
+
+                let local_send_msg = SubMsg::new(
+                    execute_on_behalf_of(vec![transfer_msg], cfg.input_addr())?
+                        .change_custom()
+                        .ok_or_else(|| {
+                            ServiceError::ExecutionError(
+                                "Failed to change local send msg custom type.".to_owned(),
+                            )
+                        })?,
+                );
```

Comment:
Each bridge might have its own rules, and encapsulating this in the service might be the best idea.
1 thing that comes with mind is how neutron ibc fee works, it is taken from the contract, or at least used to be taken from the sender contract.

On its own its not a big deal, but lets assume we have an amount sensitive workflow or service which requires for a specific amount to be handled.
If this service is working on `untrn` there will be an issue here, because the ibc fee will be taken from the account, which will result in `amount - 1`, in most cases this will be a big issue, but I think we should separate the "bridge payments" and the "working" funds of the workflow.

So for neutron IBC transfer, we should first check if the service have enough funds to pay for this fee before trying to send the funds.